### ISSUE TITLE
Unpacking a dict lines up argument with the wrong parameter

### ISSUE THREAD

**Issue by Julian** [microsoft/pyright#7802]
**Describe the bug**
This code:

```python
def f(valid: bool, comment: str | None = None):
    pass
f(**dict(valid=True))
```

produces:

```
/Users/julian/Development/bowtie/foo.py
  /Users/julian/Development/bowtie/foo.py:3:5 - error: Argument of type "bool" cannot be assigned to parameter "comment" of type "str | None" in function "f"
    Type "bool" is incompatible with type "str | None"
      "bool" is incompatible with "str"
      "bool" is incompatible with "None" (reportArgumentType)
1 error, 0 warnings, 0 informations 
```

where it seems the `valid` argument is somehow being considered as being `comment`. The same occurs with classes (dataclasses and attrs classes) which is how I originally noticed.


**VS Code extension or command-line**
```
⊙  .nox/typing/bin/pyright --version                                                                                                                                                                                                                                          julian@Airm ●
pyright 1.1.360
```

**Comment by erictraut** [comment]
The current behavior is intended. It's consistent with the way mypy works.

This is something that we may revisit in context of the official [Python typing spec](https://typing.readthedocs.io/en/latest/spec/index.html). I think one could make a reasonable argument that the current behavior is not desirable, although a change would trade off some false positives for other false negatives. The typing spec is currently silent about how this case should be handled, but it would be good to standardize the behavior. 

If you're interested in kicking off a discussion about [amending the typing spec](https://github.com/python/typing-council#decisions) to clarify the intended behavior in this case, you can do so in the [Python typing forum](https://discuss.python.org/c/typing/32). 

In the meantime, I'm going to close this as "won't fix" because it's working as intended currently. If and when the typing spec is updated to indicate a different behavior, I'll modify pyright to conform accordingly.
